### DISCUSSION TITLE
Feature request: to/from json serialization methods

### DISCUSSION THREAD

**Discussion by mikegratton** [eProsima/Fast-DDS-Gen#179] [Ideas]
For debugging apps that use IDL-derived types, it would be helpful to have to/from json methods.  The toJson() method makes it easy to inspect any particular message without having to write extra code.  The fromJson() method makes it easier to inject hand-made messages during testing, as the message content can be edited with a text editor.

**Comment by Mario-DL** [comment]
Hi @mikegratton 

I think that is a nice suggestion, thanks for the proposal ! Nonetheless I am moving this to the proper discussion forum since it is not an issue.

**Comment by mikegratton** [comment]
I've created a basic version of the toJson() here: https://github.com/mikegratton/Fast-DDS-Gen/tree/feature/json

The added method is `std::string toJson() const`

Some notes:
1. map types don't seem to work on master, so the json generation from maps isn't tested
2. For unions, I'm serializing unions using the inner active type name.

Here's an example:
```
enum Enumeration {
   First,
   Second
};

union MyUnion switch(Enumeration) {
   case First: 
       int x;
   case Second:
       double y;
};

struct MyStruct {
   MyUnion the_union
};
```
This serializes to json like
```
{ "the_union":{"x":123}}
```
where a dummy map is inserted so that both names (the_union and x) are visible.
